### ISSUE TITLE
Assertion violation at ctx_solver_simplify_tactic.cpp:141

### ISSUE THREAD

**Issue by rainoftime** [Z3Prover/z3#4230]
Hi, for the following formula,
~~~~
(set-logic QF_AUFNIRA)
(declare-const r2 Real)
(declare-const r0 Real)
(declare-const r1 Real)
(assert (= (* r0 r0 r0 r1) (- 0.0 (/ 0.0 r1) r2 (* r0 r0 r0 r1)) r0 3.0))
(check-sat-using ctx-solver-simplify)
~~~~

z3 (commit 1f15033) throws an assertion violation 
~~~~
ASSERTION VIOLATION
 File: smt/tactic/ctx_solver_simplify_tactic.cpp
 Line: 141
 UNREACHABLE CODE WAS REACHED.
 (C)ontinue, (A)bort, (S)top, (T)hrow exception, Invoke (G)DB
~~~~

**Comment by NikolajBjorner** [comment]
lemma 23 is not valid:

C:\z3\build>type lemma_23.smt2
(declare-fun r1 () Real)
(declare-fun r0 () Real)
(assert (not (= (* r0 r0 r0 r1) 0.0)))
(assert (not (= r1 1.0)))
(assert (>= (+ (* (- 1.0) r0 r0 r0 r1) r0) 0.0))
(assert (<= (+ (* (- 1.0) r0 r0 r0 r1) r0) 0.0))
(assert (not (<= r1 0.0)))
(check-sat)

z3 4230.smt2 smt.arith.dump_lemmas=true

**Comment by NikolajBjorner** [comment]
this wasn't resolved with the fix to #4246
